apps/tables/src/components/ViewTable.vue

```diff
@@ -25,6 +26,20 @@ export default {
     RoutedTableExplorer,
   },
   computed: {
+    canView() {
+      if (this.session) {
+        const isAdmin = this.session.email === "admin";
+        const roles = this.session.roles;
+        const isViewer =
+          (roles && roles.includes("Viewer")) ||
+          this.activeTable.tableType === "ONTOLOGIES";
+        const isEditor = roles && roles.includes("Editor");
```

Comment:
can do: roles?.includes(...) if roles is undefined the result of this is also undefined, which is falsy, so that works :)

Reply:
If for the first 2 assignments you do this.session?. email and this.session?.roles you can remove the outer if.

Reply:
indeed, I have cleaned this code up